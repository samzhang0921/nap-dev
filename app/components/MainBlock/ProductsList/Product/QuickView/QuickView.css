.overlay {
    background-color: rgba(51, 51, 51, 0.55);
    bottom: 0;
    left: 0;
    position: fixed;
    right: 0;
    top: 0;
    z-index: 99;
}
.wrap {
  margin: 5rem auto;
  background-color: #fff;
  width: 40%;
 text-align: center;
  box-sizing: border-box;
  padding: 3%;
}

h2 {
line-height: 26px;
}
