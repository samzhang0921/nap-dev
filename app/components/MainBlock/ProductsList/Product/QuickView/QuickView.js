import React from 'react';
import { render } from 'react-dom';
import style from './QuickView.css';
import fetch from 'isomorphic-fetch';
export default class QuickView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
        quickProduct:{}
    };
    this.showComponent = this.showComponent.bind(this);
  }
  showComponent() {
    this.props.showComponent();
    this.props.toggleImage();
   }

   componentDidMount() {
     console.log('componentDidMount,,quickProduct');
     const productID = this.props.id;
     const url = 'http://127.0.0.1:3000/api/product/'+productID;
     fetch(url).then(res => {
       return res.json()
     }).then(res => {
       this.setState({
         quickProduct: res

       })
     })
   };


  render() {
    return (
      <div className={style.overlay}>
            <div className={style.wrap}>
              <h2>{this.state.quickProduct.designer}</h2>
              <h3>{this.state.quickProduct.name}</h3>
              <h4>{this.state.quickProduct.price}</h4>
              <div className={style.close} onClick={this.showComponent}></div>
              <div>
              {!this.state.quickProduct.images?
                null :
                <img src={this.state.quickProduct.images.large} alt={this.state.quickProduct.name}/>
              }
              </div>
            </div>
       </div>
    )

  }
}
