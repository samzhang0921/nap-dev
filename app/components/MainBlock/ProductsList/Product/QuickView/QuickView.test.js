import React from 'react';
import {
  mount
} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import QuickView from './QuickView';

describe('QuickView', () => {
      const id = 11;
      const showComponent = jest.fn();
      const toggleImage = jest.fn();

            it('should match its snapshot with items', () => {

                const tree = renderer.create( < QuickView
                  id = {id}
                  showComponent = {showComponent}
                  toggleImage = {toggleImage}
                  />).toJSON();

                  expect(tree).toMatchSnapshot();
                });

              it('should change class on the onClick function', () => {

                  const component = mount( < QuickView
                    id = {id}
                    showComponent = {showComponent}
                    toggleImage = {toggleImage}
                    />);
                    component.find('.close').simulate('click'); expect(component.find('img')).toHaveLength(0);
                  });
              });
