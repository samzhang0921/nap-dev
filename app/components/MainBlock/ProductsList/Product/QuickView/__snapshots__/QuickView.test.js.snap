// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickView should match its snapshot with items 1`] = `
<div
  className="overlay"
>
  <div
    className="wrap"
  >
    <h2 />
    <h3 />
    <h4 />
    <div
      className="close"
      onClick={[Function]}
    />
    <div />
  </div>
</div>
`;
